frontend/src/data/utils/DatasetTypes.ts

```diff
@@ -59,32 +60,45 @@ export class Dataset {
   }
 
   toDataFrame(): IDataFrame {
-    // TODO Remove this once STATE FIPS are embedded during GCP/SQL step
-    const rowsWithFakeFips = this.rows.map((row) => {
-      const fipsCode = Object.keys(STATE_FIPS_MAP).find(
-        (key) => STATE_FIPS_MAP[key] === row.state_name
-      );
-      return { ...row, state_fips: fipsCode };
-    });
-    return new DataFrame(rowsWithFakeFips);
+    return new DataFrame(this.rows);
   }
 
-  toCsvString() {
-    // Assume the columns are the same as the keys of the first row. This is
-    // okay since every row has the same keys. However, we could improve this by
-    // sending column names as structured data from the server.
-
-    const fields = Object.keys(this.rows[0]);
-
-    const df = this.toDataFrame().transformSeries(
-      Object.fromEntries(
-        fields.map((name) => [name, convertSpecialCharactersForCsv])
-      )
-    );
-    return [fields]
-      .concat(df.toRows())
-      .map((row) => row.join(","))
-      .join("\n");
+  getAllColumnNames(): string[] {
+    const headersSet = new Set<string>();
+    for (const row of this.rows) {
+      for (const header of Object.keys(row)) {
+        headersSet.add(header);
+      }
+    }
+    return Array.from(headersSet);
+  }
+
+  toCsvString(): string {
```

Comment:
Super interesting, I'm surprised there isn't a built in way to do this.

Reply:
There is but it defaults to inferring the column names from the first json object/row. The API of the library they're extending in Dataforge seems to allow for passing in the column names explicitly as an argument but I wasn't able to get it working